### DISCUSSION TITLE
ReactSelect needs to be provide trigger for "native form onChange"

### DISCUSSION THREAD

**Discussion by arifemir** [JedWatson/react-select#5240] [Ideas]
**ReactSelect**  needs to be provide trigger for "native form onChange"

```
<form onChange={(e) => {console.log(e)}>
  <ReactSelect /> // this is not trigger onChange
  <input type='text' /> // this trigger onChange 
</form>
```

**Comment by gormonn** [comment]
https://github.com/JedWatson/react-select/issues/3798

**Comment by Rall3n** [comment]
@arifemir Afaik this is not possible.

**To clarify:**
`react-select` does not use a native `<select>` element, which is a reason why the component does not emit a native `onChange` event.

You can create events using `Event()` (or `CustomEvent()`) constructor and trigger them with `element.dispatchEvent()`.
But as far as I experienced these events are not always triggered. I don't have the source anymore (and the spec does not clearly say this either), but I believe browsers can ignore script-triggered events using the read-only `isTrusted` flag (default `false` using `dispatchEvent`). Only exception regarding `isTrusted` is the `click` event.
